pandas/tests/arrays/test_period.py

```diff
@@ -225,8 +225,7 @@ def test_sub_period():
 def test_where_different_freq_raises(other):
     ser = pd.Series(period_array(['2000', '2001', '2002'], freq='D'))
     cond = np.array([True, False, True])
-    with pytest.raises(IncompatibleFrequency,
-                       match="Input has different freq=H"):
+    with pytest.raises(IncompatibleFrequency, match="freq"):
```

Comment:
Needed this change, as `__setitem__` might raise when creating the PeriodArray from a PeriodArray with different freq now, with a slightly different error message.

We might want to change `validate_dtype_freq` to use the error message from `raise_on_incompatible`. I haven't looked closely.